src/integrator_interface.jl

```diff
@@ -5,13 +5,15 @@ Update solution of `integrator`, if necessary or forced by `force_save`.
 """
 function savevalues!(integrator::DDEIntegrator, force_save=false)
     # update time of ODE integrator (can be slightly modified (< 10ϵ) because of time stops)
-    if integrator.integrator.t != integrator.t
-        if abs(integrator.t - integrator.integrator.t) >= 10eps(integrator.t)
-            error("unexpected time discrepancy detected")
-        end
+    if typeof(integrator.t) <: AbstractFloat # does not work for units!
+        if integrator.integrator.t != integrator.t
+            if abs(integrator.t - integrator.integrator.t) >= 10eps(integrator.t)
```

Comment:
Could we just multiply by `oneunit(integrator.t)`? `10eps(integrator.t)*oneunit(integrator.t)` for that to work?

Reply:
No, we need something different `eps(integrator.t)` is not defined for units. So maybe something like `10eps(integrator.t/oneunit(t))*oneunit(integrator.t)`. The problem occurs in `OrdinaryDiffEq`, as well (in `integrator_utils.jl` e.g.).

Reply:
I see. Can you make it separate?

Reply:
A separate PR for OrdinaryDiffEq with this change?

Reply:
Yeah and here.